### DISCUSSION TITLE
"BT-01-notice" type "code" and invalid "listName" attribute

### DISCUSSION THREAD

**Discussion by jaggaer-fgeniola** [OP-TED/eForms-SDK#590] [Q&A] [answered]
Hi,
I read in another (old) post that fields of type "code" must have the "listName" attribute in the XML element.
Following this rule, however, an error is generated for some fields because they do not allow the aforementioned attribute, eg "BT-01-notice".
How can we identify in which fields to put the "listName" attribute and in which others not?

Thanks

**Comment by YvesJo** [comment, marked as answer]
Hi,
This is a concern also raised by other stakeholders, besides the XSD and the documentation, there is nothing really "friendly" to help detect how this should be processed.
The element carrying that information in the XML is of type "text" and therefore does not have any associated listName attribute. With SDK 1.9, implicit attributes will be explicitely identified in the field.json. In any previous version it would be needed to extract that information from: 

- the schema or 
- from the documentation and have it hardcoded

**Reply by lepo** [reply]
Hi,
Some fields has the listName information in the xPaths, others not. Maybe you are able to re-generate the XPath informations and update the fields.json.
e.g. /*/cac:ContractingParty/cac:ContractingActivity/cbc:ActivityTypeCode[@listName='authority-activity'] field BT-10-Procedure-Buyer

How this info will be coded in SDK Version 1.9.? It's a very long time to Version 1.9. and every temporary workaround costs development time.
